src/opower/utilities/exelon.py

```diff
@@ -25,11 +32,63 @@ def timezone() -> str:
     def login_domain() -> str:
         """Return the domain that hosts the login page."""
         raise NotImplementedError
+    
+    @classmethod
+    async def subdomain(cls) -> str:
+        """Return the opower.com subdomain for this utility."""
+        if Exelon._subdomain is not None:
+            return Exelon._subdomain
+        else:
+            # Get the potential url subdomains, they can vary based on account type/location
+            async with cls._session.get(
+                "https://" + cls.login_domain() + "/api/Services/MyAccountService.svc/GetConfiguration",
+                headers={"User-Agent": USER_AGENT},
+                raise_for_status=True,
+            ) as resp:
+                data = await resp.json()
+
+                opCo = data["opCo"].lower()
+                oPowerURLBase = data["oPowerURLBase"].split(".", 1)[0].replace("https://", "")
+                oPowerURLBaseJurisdiction = data["oPowerURLBaseJurisdiction"].split(".", 1)[0].replace("https://", "")
+                _LOGGER.debug("found exelon opCo: %s", opCo)
+                _LOGGER.debug("found exelon oPowerURLBase: %s", oPowerURLBase)
+                _LOGGER.debug("found exelon oPowerURLBaseJurisdiction: %s", oPowerURLBaseJurisdiction)
+
+            # Get the account type & state
+            async with cls._session.get(
+                "https://" + cls.login_domain() + "/.euapi/mobile/custom/auth/accounts",
+                headers={"User-Agent": USER_AGENT, "authorization": f"Bearer {cls._access_token}"},
+                raise_for_status=True,
+            ) as resp:
+                # returned mimetype is nonstandard, so this avoids a ContentTypeError
+                response = await resp.json(content_type=None)
+
+                #Only include active accounts (after moving, old accounts have status: "Inactive")
+                #NOTE: this logic currently assumes 1 active address per account, if multiple accounts found
+                #      we default to taking the first in the list. Future enhancement is to support
+                #      multiple accounts (which could result in different subdomain for each)
+                active_accounts = [account for account in response['data'] if account['status'] == 'Active']
+                isResidential = active_accounts[0]["isResidential"]
+                state = active_accounts[0]['PremiseInfo'][0]['mainAddress']['townDetail']['stateOrProvince']
+                _LOGGER.debug("found exelon account isResidential: %s", isResidential)
+                _LOGGER.debug("found exelon account state: %s", state)
+
+            # Determine subdomain to use by matching logic found in https://cls.login_domain()/dist/app.js
+            Exelon._subdomain = oPowerURLBase

```

Comment:
Can you initialize this in async_login instead and keep the subdomain method synchronous?